engine/src/main/java/org/terasology/engine/logic/players/LocalPlayer.java

```diff
@@ -236,24 +237,27 @@ private boolean activateTargetOrOwnedEntity(EntityRef usedOwnedEntity) {
         boolean ownedEntityUsage = usedOwnedEntity.exists();
         int activationId = nextActivationId++;
         Physics physics = CoreRegistry.get(Physics.class);
-        HitResult result = physics.rayTrace(originPos, direction, characterComponent.interactionRange,
+
+        float interactionRange;
+        if (ownedEntityUsage && usedOwnedEntity.hasComponent(RangeComponent.class)) {
+            interactionRange = Math.max(usedOwnedEntity.getComponent(RangeComponent.class).value, characterComponent.interactionRange);
+        } else {
+            interactionRange = characterComponent.interactionRange;
+        }
+
+        HitResult result = physics.rayTrace(originPos, direction, interactionRange,
                 Sets.newHashSet(character), CharacterSystem.DEFAULTPHYSICSFILTER);
         boolean eventWithTarget = result.isHit();
-        if (eventWithTarget) {
+
+        if (ownedEntityUsage || eventWithTarget) {
             EntityRef activatedObject = usedOwnedEntity.exists() ? usedOwnedEntity : result.getEntity();
             activatedObject.send(new ActivationPredicted(character, result.getEntity(), originPos, direction,
                     result.getHitPoint(), result.getHitNormal(), activationId));
+
             character.send(new ActivationRequest(character, ownedEntityUsage, usedOwnedEntity, eventWithTarget,
                     result.getEntity(),
                     originPos, direction, result.getHitPoint(), result.getHitNormal(), activationId));
             return true;
-        } else if (ownedEntityUsage) {
-            usedOwnedEntity.send(new ActivationPredicted(character, EntityRef.NULL, originPos, direction,
-                    originPos, new Vector3f(), activationId));
```

Comment:
Instead of `originPos` and `new Vector3f()` we now put `null` for both in case there is no target. This changes semantics, but I added this as docstring to at least help with debugging.